### PR TITLE
[New project structure] Light Client module

### PR DESCRIPTION
**Pull request by bullet-tooth**
## Overview

---
Created Light Client module

### Definition of Done

- [ ] There are no TODOs left in the code
- [ ] Change is covered by automated [tests](https://github.com/exonum/exonum-java-binding/blob/master/CONTRIBUTING.md#tests)
- [ ] The [coding guidelines](https://github.com/exonum/exonum-java-binding/blob/master/CONTRIBUTING.md#the-code-style) are followed
- [ ] Public API has Javadoc
- [ ] Method preconditions are checked and documented in the Javadoc of the method
- [ ] Changelog is updated if needed (in case of notable or breaking changes)
- [ ] The continuous integration build passes

### CODE REVIEW

File: run_all_tests.sh [hunk 1]
```diff
@@ -1,12 +1,29 @@
...
 #
 # A JVM will be selected by JAVA_HOME environment variable, or, if it is not set,
 # inferred from the java executable available on the path.
 
 # Fail immediately in case of errors and/or unset variables
 set -eu -o pipefail
 
-echo "Start running all EJB tests"
+# Import necessary environment variables (see the tests_profile header comment for details).
+source exonum-java-binding-parent/tests_profile
```
Comment (MakarovS) [inline run_all_tests.sh]: Shouldn't we change `source tests_profile` in other places as well, like `pom.xml`?

Comment (bullet-tooth) [reply run_all_tests.sh]: As far as I can see everything fine. `tests_profile` moved together with `pom.xml` to the exonum-java-binding-parent dir

File: exonum-java-light-client/README.md [hunk 2]
```diff
@@ -0,0 +1,30 @@
+# Exonum Java Light Client
+Java client for [Exonum blockchain][exonum].
```
Comment (dmitry-timofeev) [inline exonum-java-light-client/README.md]: Shall we put somewhere here an overview of its intended functionality (what exactly the light client is and what it allows to do)?

File: run_all_tests.sh [hunk 3]
```diff
@@ -1,12 +1,29 @@
...
+#  - Test coverage information collection.
+# See build definitions of the modules for more.
+mvn install \
+  --activate-profiles ci-build \
+  -Drust.compiler.version="${RUST_COMPILER_VERSION}"
+
+echo "Start running EJB native tests"
 cd exonum-java-binding-parent
-./run_all_tests.sh
+# Run native integration tests that require prepared classpaths for fake classes.
```
Comment (dmitry-timofeev) [inline run_all_tests.sh]: `run_all_tests.sh` in `ejb-parent/` remains as is to be able to build from that directory, is that correct?

Shall we put for now a NOTE saying to keep the two in sync?

File: exonum-java-light-client/README.md (hunk 2 above)
Comment (bullet-tooth) [reply exonum-java-light-client/README.md]: yes, sure. But I'm going to do that after P1 development, to write the readme once along with getting started guide, examples etc.

File: run_all_tests.sh (hunk 3 above)
Comment (bullet-tooth) [reply run_all_tests.sh]: Yes, it is correct.
Ok, I'll write notes explaining this stuff.

File: exonum-java-light-client/README.md (hunk 2 above)
Comment (dmitry-timofeev) [reply exonum-java-light-client/README.md]: Makes sense!

File: exonum-java-light-client/pom.xml [hunk 4]
```diff
@@ -145,4 +153,49 @@
...
+            <artifactId>jacoco-maven-plugin</artifactId>
+            <version>0.8.3</version>
+            <executions>
+              <execution>
+                <id>prepare-jacoco-agent</id>
+                <goals>
+                  <goal>prepare-agent</goal>
+                </goals>
+                <configuration>
+                  <destFile>${project.build.directory}/coverage-reports/jacoco-unit.exec</destFile>
```
Comment (dmitry-timofeev) [inline exonum-java-light-client/pom.xml]: Nit: May extract in a property like "jacoco unit coverage info path"

**Pull request merged**